actors/verifreg/src/lib.rs

```diff
@@ -406,42 +407,54 @@ impl Actor {
     {
         // since the alloc is expired this should be safe to publically cleanup
         rt.validate_immediate_caller_accept_any()?;
+        let client = params.client;
         let mut ret_gen = BatchReturnGen::new(params.allocation_ids.len());
-        rt.transaction(|st: &mut State, rt| {
-            let mut allocs = st.load_allocs(rt.store())?;
-            for alloc_id in params.allocation_ids {
-                let maybe_alloc = allocs.get(params.client, alloc_id).context_code(
-                    ExitCode::USR_ILLEGAL_STATE,
-                    "HAMT lookup failure getting allocation",
-                )?;
-                let alloc = match maybe_alloc {
-                    None => {
-                        ret_gen.add_fail(ExitCode::USR_NOT_FOUND);
-                        info!(
+        let mut recovered_datacap = DataCap::zero();
+        let token_addr = rt
+            .transaction(|st: &mut State, rt| {
+                let mut allocs = st.load_allocs(rt.store())?;
+                for alloc_id in params.allocation_ids {
+                    let maybe_alloc = allocs.get(client, alloc_id).context_code(
+                        ExitCode::USR_ILLEGAL_STATE,
+                        "HAMT lookup failure getting allocation",
+                    )?;
+                    let alloc = match maybe_alloc {
+                        None => {
+                            ret_gen.add_fail(ExitCode::USR_NOT_FOUND);
+                            info!(
                             "claim references allocation id {} that does not belong to client {}",
-                            alloc_id, params.client,
+                            alloc_id, client,
                         );
+                            continue;
+                        }
+                        Some(a) => a,
+                    };
+                    if alloc.expiration > rt.curr_epoch() {
+                        ret_gen.add_fail(ExitCode::USR_FORBIDDEN);
+                        info!("cannot revoke allocation {} that has not expired", alloc_id);
                         continue;
                     }
-                    Some(a) => a,
-                };
-                if alloc.expiration > rt.curr_epoch() {
-                    ret_gen.add_fail(ExitCode::USR_FORBIDDEN);
-                    info!("cannot revoke allocation {} that has not expired", alloc_id);
-                    continue;
+                    recovered_datacap += alloc.size.0;
```

Comment:
nit: DataCap::from(alloc)

Reply:
I deleted that `From` impl, it was too odd for me. We could do with some nicer accessors on PaddedPieceSize though.